app/src/main/java/com/tachibana/downloader/ui/main/FinishedDownloadsFragment.java

```diff
@@ -123,11 +124,13 @@ public void onSaveInstanceState(@NonNull Bundle outState)
     }
 
     @Override
-    public void onItemClicked(@NonNull DownloadItem item)
-    {
-        startActivity(Intent.createChooser(
-                Utils.createOpenFileIntent(activity.getApplicationContext(), item.info),
-                getString(R.string.open_using)));
+    public void onItemClicked(@NonNull DownloadItem item) {
+        Intent file = Utils.createOpenFileIntent(activity.getApplicationContext(), item.info);
+        if (file != null) {
+            startActivity(Intent.createChooser(file, getString(R.string.open_using)));
+        } else {
+            Toast.makeText(activity.getApplicationContext(), "file not available!", Toast.LENGTH_SHORT).show();
```

Comment:
Add a new string to `strings.xml` to localize this message.

Reply:
Done.